### PR TITLE
CEDS-2170 ile query controller fix

### PR DESCRIPTION
**Pull request by MaciejRewera**

### CODE REVIEW

File: app/connectors/CustomsDeclareExportsMovementsConnector.scala [hunk 1]
```diff
@@ -108,8 +106,8 @@ class CustomsDeclareExportsMovementsConnector @Inject()(appConfig: AppConfig, ht
     httpClient
       .GET[HttpResponse](s"${appConfig.customsDeclareExportsMovementsUrl}$IleQuery/$conversationId", providerIdQueryParam(providerId))
       .recoverWith {
-        case exception: Upstream4xxResponse if exception.upstreamResponseCode == FailedDependencyStatus =>
-          Future.successful(HttpResponse(responseStatus = FailedDependencyStatus))
+        case exception: Upstream4xxResponse if exception.upstreamResponseCode == Status.FAILED_DEPENDENCY =>
```
Comment (PauloPortugal) [inline app/connectors/CustomsDeclareExportsMovementsConnector.scala]: :+1:

File: app/models/notifications/queries/IleQueryResponseExchangeData.scala [hunk 2]
```diff
@@ -44,6 +44,16 @@ object IleQueryResponseExchangeData {
     override val typ = IleQueryResponseExchangeType.SuccessfulResponseExchange
 
     val sortedChildrenUcrs: Seq[UcrInfo] = (childDucrs ++ childMucrs).sortBy(_.entryStatus.flatMap(_.roe).getOrElse(ROECode.NoneRoe))
+
+    lazy val queriedUcr: UcrInfo = {
```
Comment (SteveSugden) [inline app/models/notifications/queries/IleQueryResponseExchangeData.scala]: You can do something like this (based on the controller code you are removing)
```
lazy val queriedUcr: UcrInfo = queriedDucr
      .orElse(queriedMucr)
      .getOrElse(throw new IllegalStateException("SuccessfulResponseExchangeData must have either queriedDucr or queriedMucr"))
```

Comment (SteveSugden) [reply app/models/notifications/queries/IleQueryResponseExchangeData.scala]: Can the `sortedChildrenUcrs` val be lazy as well?

File: app/models/notifications/queries/UcrInfo.scala [hunk 3]
```diff
@@ -24,5 +26,10 @@ abstract class UcrInfo {
   val entryStatus: Option[EntryStatus]
   val movements: Seq[MovementInfo]
 
-  def transport: Option[Transport] = movements.find(_.transportDetails.isDefined).flatMap(_.transportDetails)
+  def transport: Option[Transport] =
```
Comment (SteveSugden) [inline app/models/notifications/queries/UcrInfo.scala]: can this be a lazy val also - no need to run this expensive computation more than once.

**Pull request merged**